opesci/util.py

```diff
@@ -118,13 +118,28 @@ def Taylor_half(dx, n):
     return Matrix(l)
 
 
-def Deriv(U, i, k, d, n):
+def Taylor_generic(delta, left, right, half=False):
+    """
+    generic version of Taylor()
+    expand neighbour grid point located from -left to +right
+    if half is True, shift to right by half
+    """
+    l = []
+    n = left + right + 1
+    shift = hf if half else 0
+    for i in range(-left, right+1):
+        ll = [tc((i+shift)*delta, j) for j in range(n)]
+        l.append(ll)
+    return Matrix(l)
+
+
+def Deriv(U, index, k, d, n):
     """
     calculate the FD approximation for nth derivative
     the function works by using M * D = R, as specified in Taylor()
     then inverting M, so that D = M_inverse * R
     :param U: the field
-    :param i: list of indices (Symbol objects) of field U
+    :param index: list of indices (Symbol objects) of field U
```

Comment:
Note that you can also specify the data type in between `param` and the `param`'s name, e.g. `:param list index:`